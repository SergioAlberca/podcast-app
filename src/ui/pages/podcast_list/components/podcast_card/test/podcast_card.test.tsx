import { vi } from "vitest";
import { fireEvent, render } from "@/test/test_utils";
import PodcastCard from "../podcast_card";
import { fixtures } from "./fixtures/fixtures";

const goToDetail = vi.fn();

describe("Podcast Card component test", () => {
  it("render component", () => {
    const { container } = render(<PodcastCard podcast={fixtures} goToDetail={goToDetail} />);
    expect(container).toMatchSnapshot();
  });

  it("Author name to be in the document", () => {
    const { getByText } = render(<PodcastCard podcast={fixtures} goToDetail={goToDetail} />);
    expect(getByText(fixtures["im:name"].label)).toBeInTheDocument();
  });

  it("goToDetail has been called", () => {
    const { getByTestId } = render(<PodcastCard podcast={fixtures} goToDetail={goToDetail} />);
    fireEvent.click(getByTestId("podcast_card"));
    expect(goToDetail).toHaveBeenCalled();
  });
});
