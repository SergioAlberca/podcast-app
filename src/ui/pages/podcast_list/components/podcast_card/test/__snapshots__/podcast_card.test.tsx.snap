// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Podcast Card component test > render component 1`] = `
<div>
  <div
    aria-hidden="true"
    class="_card_250b91"
    data-testid="podcast_card"
  >
    <img
      alt="podcast_image"
      class="_card__img_250b91"
      src="https://is3-ssl.mzstatic.com/image/thumb/Podcasts113/v4/f2/21/fa/f221fabd-017f-5125-633b-f1fe4f39802a/mza_182995249085044287.jpg/55x55bb.png"
    />
    <h5>
      The Joe Budden Podcast
    </h5>
    <span
      class="_card__text_250b91"
    >
      Author: 
      The Joe Budden Network
    </span>
  </div>
</div>
`;
